### PR TITLE
Backup: Add backup storage UI

### PR DESCRIPTION
**Pull request by Initsogar**
#### Changes proposed in this Pull Request:

- [x] Add site storage size and storage limit on backup admin page
- [x] Add progress bar with different colors based on % of storage usage
- [x] Adjust new layout to include the new storage UI
- [x] When storage usage is above 65%, show a CUT with add-on storage upgrade

#### What's next?
The following features will be address in a new PR:
- Add days of backups saved below the storage meter.
- Backups stopped related messages on main title and CUT.

#### Screenshots
| Description | Screenshot |
|---|---|
| Admin page | image
| Normal (< 65% used storage) |  |
| Warning (65% - 80% used storage) |  | 
| Critical (80%-100% used storage) |  |
| Full (> 100% used storage) |  |

#### Other information:
- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Jetpack product discussion

* peaFOp-wy-p2
* p1HpG7-jQY-p2

#### Does this pull request change what data or activity we track or use?

No

#### Unit testing:
You can run unit testing using the following command:
* JS: `jetpack test packages/backup js`
* PHP: `jetpack test packages/backup php`

#### Testing instructions:

* Spin up a Jurassic Ninja site with Jetpack Beta, using Backup branch `add/backup-storage-ui` with a backup plan like Jetpack Backup (1 GB or 10 GB).
* Navigate to Jetpack -> VautlPress Backup
* Once the first backup completes, you should see the main page of the plugin, including the storage meter.
* Given it is a new site, it will be ~0GB used and no upsell prompt should be displayed.

To try different scenarios for the storage meter, you could try to upload big files to your site or dispatch the events using [Redux DevTools (chrome extension)](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en). After installing it, follow these steps:
* Open the extension in Chrome
* In the dropdown in the top right, select `jetpack-backup-plugin`
* In the bottom, click on `>_` option to show the dispatcher.
image
* It should look like this:

* Then in the dispatcher editor, you can set the used storage like this (the size is represented in bytes):
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 10737418240
    }
}
```

* So if you have a site with 10 GB, you can use the following sizes to test the different scenarios:
  * Normal (< 65%): 1073741824 (1GB)
  * Warning (65% - 80%): 7516192768 (7GB)
  * Critical (80%-100%): 9663676416 (9GB)
  * Full (> 100%): 10737418240 (10 GB)

### CODE REVIEW

**Review by keoshi — DISMISSED** [review]
First off, thank you so much for all the work that went into this, @Initsogar! The detailed instructions are wonderful as well.

Everything looks great to me in the testing that I did. Noticed some minor things which I'll list below, but none are blocking; more like nice to haves if we want to create follow-up PRs.

#### Zero character shows up — bug?

While the first backup is being created a zero (`0`) appears in the section created for the storage UI; see below:

image

#### Grid positioning

The new storage UI seems to take 6 columns of the existing grid, instead of the last 5 as intended. This one is on me for not having documented this more in detail...sorry about that.

See the following image where the top is the design and below what's in the PR right now, with a red rectangle of where the storage UI should be placed.

image

This might seem confusing or counter-intuitive, but by adding a full empty column in between the left and right sections, we're creating more opportunities to draw the eye to the storage UI.

#### Heading in warning state

Saw that you added variable headings for different states — very nice touch. The warning state is missing a custom heading though. Wonder if we could add this one:

> Cloud storage is almost full

**Comment by Initsogar** [comment]
@keoshi thank you for the valuable feedback! Here's my comments:

- [x] **Zero character shows up — bug?**
It was showing a 0 when the component wasn't rendering. Just fixed it.

- [x] **Fix grid positioning**
Thanks for clarifying, for some reason my eyes decided to omit that empty column 😅. Fixed!

- [ ] **Heading in warning state**
Currently, I have set to display `Cloud storage full` when the storage usage is 100% or over; otherwise it would be `Cloud storage space`.
I was thinking that `Cloud storage is almost full` fits better for storage usage between 80%-100% (Critical). What do you think? Maybe we can include it as well for 65%-80% (Warning) and Critical.

File: projects/packages/backup/src/class-jetpack-backup.php [hunk 1]
```diff
@@ -480,6 +524,142 @@ public static function manage_dismissed_backup_review_request( $request ) {
...
+	 * Get suggested storage addon based on storage usage
+	 *
+	 * @param int $bytes_used      Storage used.
+	 * @param int $bytes_available Storage limit.
+	 * @return string Suggested addon storage slug
+	 */
+	public static function get_storage_addon_upsell_slug( $bytes_used, $bytes_available ) {
+		$bytes_10gb  = 10737418240;
+		$bytes_100gb = 107374182400;
+		$bytes_1tb   = 1099511627776;
```
Comment (macbre) [inline projects/packages/backup/src/class-jetpack-backup.php]: Can we leave a comment here explaining where these IDs are coming from?

File: projects/packages/backup/src/class-jetpack-backup.php [hunk 2]
```diff
@@ -480,6 +524,142 @@ public static function manage_dismissed_backup_review_request( $request ) {
...
+	}
+
+	/**
+	 * Get suggested storage addon based on storage usage
+	 *
+	 * @param int $bytes_used      Storage used.
+	 * @param int $bytes_available Storage limit.
+	 * @return string Suggested addon storage slug
+	 */
+	public static function get_storage_addon_upsell_slug( $bytes_used, $bytes_available ) {
```
Comment (macbre) [inline projects/packages/backup/src/class-jetpack-backup.php:585]: Some tests coverage would be nice here 🙂

File: projects/packages/backup/src/class-jetpack-backup.php (hunk 1 above)
Comment (pypt) [reply projects/packages/backup/src/class-jetpack-backup.php]: I think those are not IDs but rather byte sizes. One way to remake this into a more readable form would be:

```php
		$bytes_10gb  = 10 * 1024 * 1024 * 1024;
		$bytes_100gb = 100 * 1024 * 1024 * 1024;
		$bytes_1tb   = 1024 * 1024 * 1024 * 1024;
```

Comment (macbre) [reply projects/packages/backup/src/class-jetpack-backup.php]: You're right 🤦🏻  But still, worth making these more readable.

**Comment by keoshi** [comment]
> I was thinking that Cloud storage is almost full fits better for storage usage between 80%-100% (Critical).

@Initsogar This sounds wonderful to me!

**Comment by Initsogar** [comment]
> > I was thinking that Cloud storage is almost full fits better for storage usage between 80%-100% (Critical).
> 
> @Initsogar This sounds wonderful to me!

@keoshi Great! Thanks for the feedback.

- [x] **Heading in warning state**
Just added the `Cloud storage is almost full` when the storage usage is 80%-100% (Critical).

image

File: projects/packages/backup/src/class-jetpack-backup.php (hunk 2 above)
Comment (Initsogar) [reply projects/packages/backup/src/class-jetpack-backup.php:585]: @macbre I just added some tests for this function.
Could you [take a look](https://github.com/Automattic/jetpack/pull/28085/files#diff-f0772d597476355e68faa1da0c2f88871f9856040fe8d3157711ec2021c353d9) and let me know what do you think?

File: projects/packages/backup/src/class-jetpack-backup.php (hunk 1 above)
Comment (Initsogar) [reply projects/packages/backup/src/class-jetpack-backup.php]: Thanks @macbre and @pypt for the recommendation.
Just made it a bit clearer, could you [take a look](https://github.com/Automattic/jetpack/pull/28085/files#diff-d8bc480adb260c41153b4d6a0614a3b5e8eb0b98cc6a151b19aa308c66ccaa05R586)?

File: projects/packages/backup/src/class-jetpack-backup.php (hunk 2 above)
Comment (macbre) [reply projects/packages/backup/src/class-jetpack-backup.php:585]: Perfect! 👌🏻

File: projects/packages/backup/tests/php/test-storage-addon-upsell.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+use PHPUnit\Framework\TestCase;
+
+require_once __DIR__ . '/../../src/class-jetpack-backup.php';
+
+/**
+ * Unit tests for storage addon fetch
+ *
+ * @package automattic/jetpack-backup
+ */
+class Test_Storage_Addon_Upsell extends TestCase {
```
Comment (macbre) [inline projects/packages/backup/tests/php/test-storage-addon-upsell.php:14]: Thanks for adding the test suite.

File: projects/packages/backup/src/class-jetpack-backup.php (hunk 1 above)
Comment (macbre) [reply projects/packages/backup/src/class-jetpack-backup.php]: Way better now!

File: projects/packages/backup/src/js/components/backup-storage-space/index.jsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState, useEffect, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import React from 'react';
```
Comment (jwebbdev) [inline projects/packages/backup/src/js/components/backup-storage-space/index.jsx]: Do we need to import React here?

File: projects/packages/backup/src/js/components/backup-storage-space/storage-usage-details/use-storage-usage-text.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import { useMemo, createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import React from 'react';
```
Comment (jwebbdev) [inline projects/packages/backup/src/js/components/backup-storage-space/storage-usage-details/use-storage-usage-text.tsx]: Do we need to import React here?

File: projects/packages/backup/src/js/components/backup-storage-space/index.jsx (hunk 4 above)
Comment (Initsogar) [reply projects/packages/backup/src/js/components/backup-storage-space/index.jsx]: @jwebbdev No, thanks for catching it! Just removed it.

File: projects/packages/backup/src/js/components/backup-storage-space/storage-usage-details/use-storage-usage-text.tsx (hunk 5 above)
Comment (Initsogar) [reply projects/packages/backup/src/js/components/backup-storage-space/storage-usage-details/use-storage-usage-text.tsx]: Same as above, just removed it :)

**Review by jwebbdev — DISMISSED** [review]
Looks good to me!

File: projects/packages/backup/package.json [hunk 6]
```diff
@@ -26,7 +26,9 @@
 		"extends @wordpress/browserslist-config"
 	],
 	"dependencies": {
+		"@automattic/components": "2.0.1",
 		"@automattic/jetpack-analytics": "workspace:*",
+		"@automattic/format-currency": "1.0.1",
```
Comment (anomiex) [inline projects/packages/backup/package.json]: Alphabetical order is nice to maintain.
```suggestion
		"@automattic/components": "2.0.1",
		"@automattic/format-currency": "1.0.1",
		"@automattic/jetpack-analytics": "workspace:*",
```

**Review by anomiex — COMMENTED** [review]
One minor suggestion inline. Otherwise looks ok to me from a monorepo perspective.

File: projects/packages/backup/package.json (hunk 6 above)
Comment (Initsogar) [reply projects/packages/backup/package.json]: Thanks for the suggestion! Just commited it 😊

**Review by jwebbdev — APPROVED** [review]

**Pull request merged**